Add June 25th 2026 TSC mtg notes (#256)
dreamworksanimation/openmoonray: tsc/meetings/2026/2026-06-25.md

# MoonRay TSC Meeting - June 25, 2026

## Quick Recap

> This TSC meeting focused on CI progress updates and versioning scheme discussions for OpenMoonRay. Randy, covering for Jon who was out on vacation, provided updates on DreamWorks' progress with moving to ASWIF and JFK, including initial contributions to the ASWF-Docker repo with Conan recipes for Moonray-specific dependencies like Embree and OpenImageDenoise, and testing builds for VFX Reference Platforms 2023-2025. Jean-Francois discussed his work on releasing updated container images including 2027 pre-release images based on Rocky9, noting challenges with pre-built binaries for ISPC and OpenImageDenoise due to hard-coded RPaths and TBB version conflicts. The team extensively discussed versioning schemes, with consensus forming around a date-based system using ISO format (year-week-calendar week) with decimal separators instead of the ISO standard letter, and potentially including patch versions. The group explored options for running regression tests (RATS) in the ASWF environment, discussing the limitations of GitHub Actions free runners versus paid runners, and considering alternatives like AWS CodeBuild or third-party CI hardware providers. The conversation ended with a discussion about Jakub's contributions for Houdini integration, where Rob explained they would need to break the large pull request into smaller, more atomic changes, though they would credit Jakub for his work on the Hydra 2 implementation.

Reviewer: ```suggestion
The meeting focused on CI progress updates and versioning scheme discussions for MoonRay.   Randy also provided updates on DreamWorks' progress moving to ASWF.

On the CI front, J-F Panisset has made initial contributions to the ASWF-Docker repo with Conan recipes for MoonRay-specific dependencies like Embree and OpenImageDenoise (OIDN), and is testing builds for VFX Reference Platforms 2023-2025.

J-F is also working on pre-release images for refplat 2027, based on Rocky9 and Cuda 13 and should be done by end of week.  This will include moonray-ci images.  He also notes that we can pull pre-built Conan packages if we want.  Furthermore, he notes a couple of challenges with ISPC and OIDN due to hard-coded RPaths and TBB version conflicts, that are pulling pre-built binaries from Intel's GitHub.   He suggests we could try an older version of OIDN to see if that works.

On the versioning front, the team had a good discussion on various versioning schemes, with consensus forming around a date-based system using ISO format (year-week-calendar week) with decimal separators instead of the ISO standard letter, and potentially including patch versions.  E.g. YYYY.WW.X (Year, CW, Patch).  J-F suggested that we're diligent on formatting, for instance, choosing if we'll zero pad digits so that calendar weeks would be 01 and 52, for example.   It was also brought up to perhaps add a leading single digit that indicates the versioning system in use, which is similar to what Pixar does with USD.  E.g. X.YYY.WW.X (Versioning system, Year, CW, Patch).   This was not a hard requirement, and may be overkill in our use case.

The discussion also started exploring options for running regression tests (RATS) in the ASWF environment, noting the limitations of GitHub Actions free runners versus paid runners, and considering alternatives like AWS CodeBuild or third-party CI hardware providers.  

The conversation ended with a discussion about Jakub's contributions for Houdini integration, where Rob explained they would need to break the large pull request into smaller, more atomic changes, and we would definitely credit Jakub for his work on the Hydra 2 implementation.
```

## Next Steps

> - Randy: Take notes from the CI discussion (including CUDA focus and OpenImageDenoise issues) back to Joyn.

Reviewer: ```suggestion
```

> - Randy: Propose the new OpenMoonRay versioning scheme (date-based, e.g., 2026.Wxx or 0.year.week.patch) on the Slack channel, possibly with two options for feedback.

Reviewer: ```suggestion
- Randy:  propose the new  versioning scheme (date-based, e.g., YYYY.CW.X)  on the Slack channel for feedback.
```

> - Randy: Follow up on regression testing (RATS) considerations, including exploring basic deployment on GitHub Actions free runners and assessing compute requirements.

Reviewer: ```suggestion
- Randy:  follow up with team on regression testing (RATS) considerations, including exploring basic deployment on GitHub Actions free runners and assessing compute requirements.
```

> - Randy: Coordinate with Rob and team to figure out the best forum/channel for discussing HD Moonray/Hydra 2 integration, likely in Slack or discussion forums.

Reviewer: ```suggestion
- Randy:  coordinate with Rob and team to figure out the best forum/channel for discussing  hdMoonray/Hydra 2 integration.
```

- Jean-Francois: Release a full set of updated ASWF Docker images (2023-2027, including ci-moonray) by the end of the weekend.

> - Jean-Francois: Convert ISPC Conan recipe to build from source (not pre-built binaries) in the next set of updates.

Reviewer: ```suggestion
- Jean-Francois:  convert the ISPC Conan recipe to build from source (not pre-built binaries) in the next set of updates.
```

> - Jean-Francois: Investigate using an older version of OpenImageDenoise for 2023/2024 environments due to TBB version incompatibility.

Reviewer: ```suggestion
- Jean-Francois:  investigate using an older version of OpenImageDenoise for 2023/2024 environments due to TBB version incompatibility.
```

- Toshi: Work on unit tests to make them less sensitive to the number of CPU cores, potentially by dynamically adjusting thread counts based on the environment.

> - Rob: Find the quickest way to release the new Hydra 2 (internally version 10) code to the public, possibly on a branch, and credit Jakub for contributions.

Reviewer: ```suggestion
- Rob:  determine a path to release the new Hydra 2 code to the public, possibly on a branch
```

> - DreamWorks Team: Prepare to release a new version of OpenMoonRay before SIGGRAPH.

Reviewer: ```suggestion
- OMR team: prepare to release a new version of OpenMoonRay before SIGGRAPH.
```

## Summary
### OpenMoonRay CI Progress Updates:
Randy led the meeting in place of Jon who was on vacation. He provided updates on CI progress for OpenMoonRay, including initial contributions to the ASWF-Docker repo with Conan recipes for Moonray-specific dependencies and testing builds for VFX Reference Platforms. The team plans to release these builds to OpenMoonRay after further testing, including a Mac build, with GitHub actions to build a Moonray Docker container and unit/smoke tests, aiming for a go-live by the end of the month.

### RefPlat Container Image Updates:

> Jean-Francois reported progress on releasing updated container images for RevPlat 2027, including CUDA 13 and Rocky9, with plans to complete the full set by the end of the weekend. He identified challenges with ISPC and OpenImageDenoise dependencies, particularly regarding libTBB version compatibility and hard-coded RPaths in Intel-provided binaries. The team confirmed that CUDA is the primary GPU API expected by Moonray, with CPU support as a secondary requirement. Randy announced a Birds of a Feather session scheduled for July 20th during Open Source Days, and introduced a proposal for a date-based versioning scheme using the ISO format for OpenMoonRay releases, moving away from the current divergent versioning scheme.

Reviewer: ```suggestion
Jean-Francois reported progress on releasing updated container images for RevPlat 2027, including CUDA 13 and Rocky9, with plans to complete the full set by the end of the weekend. He identified challenges with ISPC and OpenImageDenoise dependencies, particularly regarding libTBB version compatibility and hard-coded RPaths in Intel-provided binaries. The team confirmed that CUDA is the primary GPU API expected by Moonray, with CPU support as a secondary requirement. Randy announced a Birds of a Feather session scheduled for July 20th during Open Source Days, and introduced a proposal for a date-based versioning scheme using the ISO format for OpenMoonRay releases, moving away from the current, internally-divergent versioning scheme.
```
